plugins/modules/discovery_intent.py

```diff
@@ -490,6 +601,164 @@ def get_creds_ids_list(self):
         self.log("Credential Ids list passed is {0}".format(str(self.creds_ids_list)), "INFO")
         return self.creds_ids_list
 
+    def handle_global_credentials(self, response=None):
+        """
+        Method to convert values for create_params API when global paramters
+        are passed as input.
+
+        Parameters:
+            - response: The response collected from the get_all_global_credentials_v2 API
+
+        Returns:
+            - global_cred_dict  : The dictionary containing list of IDs of various types of
+                                    Global credentials.
+        """
+
+        global_credentials = self.validated_config[0].get("global_credentials")
+        cli_credentials_list = global_credentials.get('cli_credentials_list')
+        http_read_credential_list = global_credentials.get('http_read_credential_list')
+        http_write_credential_list = global_credentials.get('http_write_credential_list')
+        snmp_v2_read_credential_list = global_credentials.get('snmp_v2_read_credential_list')
+        snmp_v2_write_credential_list = global_credentials.get('snmp_v2_write_credential_list')
+        snmp_v3_credential_list = global_credentials.get('snmp_v3_credential_list')
+        net_conf_port_list = global_credentials.get('net_conf_port_list')
+        global_cred_dict = {}
+        if cli_credentials_list:
+            if not isinstance(cli_credentials_list, list):
+                msg = "Global ClI credentials must be passed as a list"
+                self.discovery_specific_cred_failure(msg=msg)
+            if len(cli_credentials_list) > 0:
+                global_cred_dict["cliCredential"] = []
+                cred_len = len(cli_credentials_list)
+                if cred_len > 5:
+                    cred_len = 5
+                for cli_cred in cli_credentials_list:
+                    if cli_cred.get('description') and cli_cred.get('username'):
+                        for cli in response.get("cliCredential"):
+                            if cli.get("description") == cli_cred.get('description') and cli.get("username") == cli_cred.get('username'):
+                                global_cred_dict["cliCredential"].append(cli.get("id"))
+                        global_cred_dict["cliCredential"] = global_cred_dict["cliCredential"][:cred_len]
+                    else:
+                        msg = "Please provide description and username of the Global CLI credential to be used"
```

Comment:
Kindly ensure you include both the description and the username for the Global CLI credential to discover the devices